app/page/info.tsx

```diff
@@ -5,120 +5,203 @@ import { Link } from "wouter"
       const balance = await provider.getBalance(account.address)
+      setBalanceLoading(false)
       setBalance(balance)
+    }
+    const id = setInterval(() => {
+      asyncIntervalFn().catch((e) => console.log(e))
+    }, 3333) // Every 3.333 seconds
 
-      const txHashes = await accountTransactions(explorer, account.address)
-      setTransactionHashes(txHashes)
+    return () => {
+      clearInterval(id)
+    }
+  }, [])
+
+  useEffect(() => {
+    // Update userOpsData when userOpStmts change
+    const userOpsData = getUserOpsData(userOpStmts, account.chainId)
+    setUserOpsData(userOpsData)
+  }, [userOpStmts])
 
-      const internalTxHashes = await accountInternalTransactions(
-        explorer,
-        account.address
+  useEffect(() => {
+    // Update node and bundler when network change
+    setNode(new NodeProvider(network.nodeRpcUrl))
+    setBundler(
+      new BundlerProvider(
+        network.bundlerRpcUrl,
+        network.chainId === 1337 ? BundlerMode.Manual : BundlerMode.Auto // 1337 is local testnet
       )
-      setInternalTransactionHashes(internalTxHashes)
+    )
+  }, [network])
+
+  useEffect(() => {
+    // Check if the broadcasted userOp has been executed successfully or unsuccessfully when sentUserOpStmts changes
+    const asyncFn = async () => {
+      sentUserOpStmts.forEach(async (sentUserOpStmt) => {
+        const userOp = new UserOperation(sentUserOpStmt.userOp)
+        const userOpHash = userOp.hash(
+          sentUserOpStmt.entryPointAddress,
+          account.chainId
+        )
+        const transactionHash = await bundler.wait(userOpHash)
+        const receipt = await node.getTransactionReceipt(transactionHash)
+
+        let status = -1
+        let blockHash = ""
+        let blockNumber = ""
+        let errorMessage = ""
+        if (receipt) {
+          //   const call = await node.call(receipt)
+          //   console.log(`[tttttt] call: ${call}`)
+          status = receipt.status
+          blockHash = receipt.blockHash
+          blockNumber = number.toHex(receipt.blockNumber)
+          errorMessage = (await bundler.getUserOperationReceipt(userOpHash))
+            .reason
+        }
+
+        switch (status) {
+          // '1' indicates a success, '0' indicates a revert, '-1' indicates a null.
+          case 0:
+            markUserOperationFailed(sentUserOpStmt.id, userOp.data(), {
+              userOpHash: userOpHash,
+              transactionHash: transactionHash,
+              blockHash: blockHash,
+              blockNumber: blockNumber,
+              errorMessage: errorMessage
+            })
+            break
+          case 1:
+            markUserOperationSucceeded(sentUserOpStmt.id, userOp.data(), {
+              userOpHash: userOpHash,
+              transactionHash: transactionHash,
+              blockHash: blockHash,
+              blockNumber: blockNumber
+            })
+            break
+          default:
+            break
+        }
+      })
```

Comment:
Status updating would not be ideal to be implemented in specific page, that's because if user never browses that page, the status of user ops in storage would never be updated. 

It could be better to have a standalone component to do this job, and pages only stays sync to those information they are interested.

Reply:
Let me consider how to present this standalone component...